api/services/S3Helper.js

```diff
@@ -43,111 +21,113 @@ class S3Client {
 
   waitForCredentials() {
     const { bucket, client } = this;
-    return retry(() => client.headBucket({ Bucket: bucket }).promise(), { waitTime: 2000 });
+    const command = new HeadBucketCommand({ Bucket: bucket });
+    return retry(async () => client.send(command), { waitTime: 2000 });
   }
 
-  listCommonPrefixes(prefix) {
+  async listCommonPrefixes(prefix) {
```

Comment:
the three list/paginated functions could go back to using a helper function like before because they all have the same structure

Reply:
OK, this is pushed.